Docker: Add troubleshooting section for Jurassic Tube cURL error

DevinWalker opened the pull request:
## Description
This PR adds a troubleshooting subsection to the Jurassic Tube documentation in the Docker README.

## Changes
- Added a new 'Troubleshooting Jurassic Tube' subsection documenting how to resolve cURL error 35 (SSL_ERROR_SYSCALL) when connecting Jetpack to WordPress.com
- Provides two solutions:
  1. Updating Docker's proxy settings through the Docker Desktop UI
  2. Adding proxy configuration constants to wp-config.php
- Includes instructions to restart Docker containers after making changes

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
- Slack thread I started: p1762822694261659-slack-CDLH4C1UZ 
- Slack thread Nick Pagazani started: p1754045486753419-slack-C05Q5HSS013
- P2 Post: pdWQjU-Ka-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:
- Review newly added section within readme for typos or potential confusing language

## Context
Automatticians sometimes encounter SSL connection errors when using Jurassic Tube to connect Jetpack to WordPress.com. This documentation helps resolve those issues quickly.

DevinWalker committed:
Docker: Add troubleshooting section for Jurassic Tube cURL error

Add documentation for resolving cURL error 35 (SSL_ERROR_SYSCALL) when Automatticians connect Jetpack to WordPress.com via Jurassic Tube.

The new troubleshooting subsection provides two solutions:
1. Updating Docker's proxy settings through Docker Desktop UI
2. Adding proxy configuration to wp-config.php

Includes reminder to restart Docker containers after making changes.

DevinWalker committed:
Update troubleshooting section for Jurassic Tube cURL error documentation

Refine the instructions for resolving cURL error 35 when connecting Jetpack to WordPress.com. Change "Solution" to "Method" for clarity and enhance the steps for configuring Docker Desktop proxy settings and wp-config.php. Include a reminder to restart Docker containers after applying changes.

jeherve reviewed (commented):
Thanks for adding this in. This may be an opportunity to update and consolidate the Jurassic Tube instructions in this repo:

- The existing section in the file you edited mentions and links to the deprecated, Docker-included version of Jurassic Tube: https://github.com/Automattic/jetpack/blob/trunk/tools/docker/README.md#jurassic-tube-tunneling-service 
It would be best to link to the standalone version of Jurassic Tube instead, like we do here: https://github.com/Automattic/jetpack/blob/trunk/docs/quick-start.md#setting-up-jurassic-tube
- Once we've done that, maybe we can shorten the section in `docs/quick-start.md` and link to the longer section in `trunk/tools/docker/README.md`?
- What would you think about shortening the section you added and just recommend the settings for everyone using Jurassic Tube, even if they don't run into the issue?

For a while, I was considering adding the 2 constants to everyone's `wp-config.php` file, but that does cause some issues in some environments (folks not using Autoproxxy, or folks working on Jetpack Boost development and working with a Boost Cloud dev environment next to the Jetpack environment) so I'm afraid it cannot be something we enable for everyone.

tbradsha commented on tools/docker/README.md:
````diff
@@ -296,6 +296,37 @@ That will let you omit those parameters while initiating the connection:
+
+```bash
+jetpack docker down
+jetpack docker up
````
I'll suggest starting it in daemon mode:

```suggestion
jetpack docker up -d
```

tbradsha reviewed (dismissed):
One inline comment, but no blockers. Thanks!

tbradsha commented on tools/docker/README.md:
```diff
@@ -296,6 +296,38 @@ That will let you omit those parameters while initiating the connection:
 
+### Troubleshooting Jurassic Tube
+
+If you're an Automattician and when connecting Jetpack to WordPress.com you receive the cURL error:
```
Either works for me.

tbradsha commented on tools/docker/README.md:
```diff
@@ -296,6 +296,38 @@ That will let you omit those parameters while initiating the connection:
+2. Click the gear icon in the top right
+3. Click the "Resources" tab
+4. Click the "Proxy" tab
+5. Toggle "Manual proxy configuration"
```
I suppose this suggestion is okay.

DevinWalker replied:
Cool 👍

tbradsha commented on tools/docker/README.md:
````diff
@@ -296,6 +296,38 @@ That will let you omit those parameters while initiating the connection:
+define( 'WP_PROXY_PORT', '8080' );
+```
+
+**Note:** Be sure to restart the Docker container after making the changes:
````
It's one parent container, so singular is fine.

DevinWalker committed:
Update tools/docker/README.md

Applying suggested edit

Co-authored-by: tbradsha <32492176+tbradsha@users.noreply.github.com>

DevinWalker committed:
Update tools/docker/README.md to correct wording on Docker container restart instruction

DevinWalker committed:
Update and consolidate Jurassic Tube documentation

- Remove deprecated Docker-included Jurassic Tube instructions
- Link to standalone Jurassic Tube setup instructions (PCYsg-GJ2-p2)
- Reframe proxy configuration as recommended settings rather than just troubleshooting
- Add note about when proxy settings may cause issues (non-Autoproxxy users, Boost developers)
- Make tools/docker/README.md the comprehensive Jurassic Tube guide
- Shorten docs/quick-start.md section and link to detailed Docker documentation
- Improves clarity and reduces documentation duplication

DevinWalker committed:
Update quick-start.md to move A12s-only note to top of section

jeherve reviewed (approved):
This is looking good now, it should be good to go! 🚢
